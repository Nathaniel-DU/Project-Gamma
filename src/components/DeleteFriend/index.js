import React, {Component} from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUserMinus } from '@fortawesome/free-solid-svg-icons';
import axios from 'axios';

export default class DeleteFriend extends Component {

    deleteUser = () => {
        axios.get(`/user/remove/${this.props.userId}`)
    }

    render(){

        return (
            <button onClick={this.deleteUser}>
                <FontAwesomeIcon className="fa-icon" icon={faUserMinus} />
            </button>
        )
    }

}