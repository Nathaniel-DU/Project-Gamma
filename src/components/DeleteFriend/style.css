.delete-icon {
    color: #ADEFD1FF;
    margin: auto!important;
}

.delete-btn {
    background-color: #00203FFF;
    border-radius: 4px;
    float: right;
    color: #ADEFD1FF;
}

.close-mdl {
    background-color: #00203FFF;
    color: white;
    border-radius: 15px;
}

.confirm-delete-btn {
    background-color: #ADEFD1FF;
    color: black;
    border-radius: 15px;
}

.modal-body, .modal-header {
    color: #00203FFF;
}