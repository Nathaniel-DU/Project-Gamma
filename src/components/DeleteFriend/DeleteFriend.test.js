import React from 'react';
import { shallow } from 'enzyme';
import DeleteFriend from './index';

describe(`DeleteFriend`, () => {
    it(`should render correctly in "debug" mode`, () => {
        const component = shallow(<DeleteFriend debug />);
        expect(component).toMatchSnapshot();
    });
});