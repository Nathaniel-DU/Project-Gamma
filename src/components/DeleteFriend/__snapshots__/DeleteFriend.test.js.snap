// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFriend should render correctly in "debug" mode 1`] = `
<be
  fluid={false}
  tag="div"
>
  <qe
    className="delete-btn"
    color="default"
    onClick={[Function]}
    size="sm"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className="fa-icon delete-icon"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            352,
            512,
            Array [],
            "f00d",
            "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
          ],
          "iconName": "times",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </qe>
  <o
    autoFocus={true}
    backdrop={true}
    backdropTransitionTimeout={150}
    centered={true}
    disableBackdrop={false}
    disableFocusTrap={true}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalStylesWithoutBackdrop={
      Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    modalTransitionTimeout={300}
    noClickableBodyWithoutBackdrop={false}
    overflowScroll={true}
    position=""
    role="dialog"
    tabIndex="-1"
    toggle={[Function]}
    zIndex={1050}
  >
    <tt
      className="modal-header"
      closeAriaLabel="Close"
      tag="h4"
      toggle={[Function]}
    >
      Confirm Deletion
    </tt>
    <$e
      className="modal-body"
    >
      Are you sure you want to remove this friend from your friend's list?
    </$e>
    <et>
      <qe
        className="close-mdl"
        color="default"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </qe>
      <qe
        className="confirm-delete-btn"
        color="default"
        onClick={[Function]}
        tag="button"
      >
        Confirm
      </qe>
    </et>
  </o>
</be>
`;
